lib/rspec/hive/query_builder.rb

```diff
@@ -0,0 +1,67 @@
+require_relative 'query_builder/row_transformer'
+require_relative 'query_builder/null_strategy'
+require_relative 'query_builder/value_by_type_strategy'
+
+module RSpec
+  module Hive
+    class QueryBuilder
+      def initialize(schema, connection)
+        @schema = schema
+        @connection = connection
+        @partition_hash = {}
+        @rows = []
+        @stubbing = false
+      end
+
+      def partition(hash)
+        partition_hash.merge!(hash)
+        self
```

Comment:
@mknapik Since we do work on an instance of the query, I change the `dup.tap { |builder| #do something with the copy }` into making the change and returning self. This way we can still chain the methods as before, but we skip creating new objects every time a method is called.